c7n/resources/s3.py

```diff
@@ -1171,26 +1171,33 @@ class ToggleVersioning(BucketActionBase):
         enabled={'type': 'boolean'})
     permissions = ("s3:PutBucketVersioning",)
 
+    def process_versioning(self, resource, state):
+        client = bucket_client(
```

Comment:
client creation should be hoisted up with client as a param here, instead of creating a client per bucket.